### PR TITLE
feat: add workerize for cleaner interop with workers

### PR DESCRIPTION
**Pull request by christjt**
# Description

Adds workerize-loader as successor to worker-loader. This adds webpack 5 support as well as adds a small RPC layer on top of loaded workers and makes communication with workers a lot easier.

See: https://github.com/developit/workerize-loader for more info.

Shortcomings / workarounds:
* Workerize loader / webpack still produces a non-inline version of the worker when building, which we don't need. So I added a webpack plugin to delete any free floating workers post-build. (added an issue here: https://github.com/developit/workerize-loader/issues/128)
* Memory transfer to / from workers are initially not supported by workerize-loader, a little library was made for this however(https://github.com/naoak/workerize-transferable). This is written and compiled to ESM, which jest didn't like (I tried for a long time to configure Jest to consume this, without any progress), so I ended up consuming the lib as part of our utilities. (here is tracking issue for transferable support directly in workerize-loader: https://github.com/developit/workerize-loader/issues/51)

# Checklist:

Here is a checklist that should completed before merging this given feature. 
Any shortcomings from the items below should be explained and detailed within the contents of this PR.

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/tsconfig.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "module": "esnext",
     "lib": [
       "dom",
-      "esnext"
+      "esnext",
+      "webworker"
```
Comment (haakonflatval-cognite) [inline viewer/tsconfig.json:9]: Was this one needed, in the end?

**Review by haakonflatval-cognite — CHANGES_REQUESTED** [review]
Looks good - a couple of things I'd like to have addressed

File: viewer/tsconfig.json (hunk 1 above)
Comment (christjt) [reply viewer/tsconfig.json:9]: Yup, needed for this specific overload apparently: https://github.com/cognitedata/reveal/pull/2321/files#diff-18839a6b13a7d5cc46b9f811597012616d27bcb330c9391ccaba84eab0d9390cR33

File: viewer/packages/utilities/src/workers/workerize-transferable/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
```
[the remark this answers is not in the document]
Comment (christjt) [reply viewer/packages/utilities/src/workers/workerize-transferable/index.ts]: fixes

File: viewer/webpack.config.js [hunk 3]
```diff
@@ -132,6 +133,19 @@ const defaultExportFunction = (env, entries, additionalAllow = undefined) => {
...
+        after: {
+          test: [
+            {
+              folder: 'dist',
+              method: absoluteItemPath => {
+                return new RegExp(/\.worker.js$/, 'm').test(absoluteItemPath);
+              }
+            }
+          ]
+          // parameters for "after normal and watch compilation" stage.
```
[the remark this answers is not in the document]
Comment (christjt) [reply viewer/webpack.config.js]: done

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

**Pull request merged**